controllers/channel.js

```diff
@@ -1,44 +1,51 @@
 'use strict';
 
 var config = require(__dirname + '/../config/config'),
-    mysql = require('anytv-node-mysql');
+    mysql = require('anytv-node-mysql'),
+    util = require(__dirname + '/../helpers/util');
 
 exports.get_channels = function (req, res, next) {
-    var limit = 50,
-        offset = 0,
+    var offset = 0,
         total = 0,
-        page = 1,
+        data = util.get_data([], ['page', 'limit', 'q'], req.query);
 
-    start = function () {
-        var query = '';
+    function start () {
+        var query = 'SELECT COUNT(channel_id) AS aggregate'
+                    + ' FROM mcn_channels'
+                    + ' WHERE (linked_date != "" OR linked_date IS NOT null)'
+                    + ' AND network != "" AND is_terminated != 1';
 
-        if (+req.query.page && +req.query.page > 0) {
-            page = +req.query.page;
-        }
+        data.page = (+data.page > 0) ? +data.page: 1;
+        data.limit = (+data.limit < 51 && +data.limit) ? +data.limit: 50;
```

Comment:
`(+data.limit < 51) ? ...`

I think that would suffice just like what you did on [line 18](https://github.com/anyTV/channel-growth-feature/pull/3/files#diff-dc6ed861d77b3279c3e1c25809b15c93R18)